src/lib/units/offset.js

```diff
@@ -149,28 +147,28 @@ export function setOffsetToUTC (keepLocalTime) {
 }
 
 export function setOffsetToLocal (keepLocalTime) {
-    if (this._isUTC) {
-        this.utcOffset(0, keepLocalTime);
-        this._isUTC = false;
+    var ret = this;
+    if (ret._isUTC) {
+        ret = ret.utcOffset(0, keepLocalTime);
+        ret._isUTC = false;
 
         if (keepLocalTime) {
-            this.subtract(getDateOffset(this), 'm');
+            ret = ret.subtract(getDateOffset(ret), 'm');
```

Comment:
If we'd done our original plan of wrapping mutable internals, then we wouldn't have intermediate copies within the library implementation.  But this comment is outdated -- we decided to work toward immutable internals instead, to make future development more understandable.